<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ReadersWindow</class>
 <widget class="QWidget" name="ReadersWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>读者管理</string>
  </property>
  <property name="whatsThis">
   <string>查看所有的读者信息，并可以修改。</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="centralLayout">
     <item>
      <widget class="QWidget" name="ReaderDetailsWidget" native="true">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>250</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>250</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">#ReaderDetailsWidget
{
	background-color: white;
	border: 1px solid rgb(235,235,235);
}
QLabel, QLineEdit
{
	font: 14px &quot;宋体&quot;;
}
QLineEdit
{
	border:0;
	border-bottom: 2px solid rgb(220,220,220);
}</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>15</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>15</number>
        </property>
        <property name="bottomMargin">
         <number>15</number>
        </property>
        <item>
         <widget class="QWidget" name="ReaderDetailsTitleWidget" native="true">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>6</number>
           </property>
           <property name="leftMargin">
            <number>9</number>
           </property>
           <property name="topMargin">
            <number>9</number>
           </property>
           <property name="rightMargin">
            <number>9</number>
           </property>
           <property name="bottomMargin">
            <number>9</number>
           </property>
           <item>
            <widget class="QLabel" name="ReaderDetailsTitle">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="styleSheet">
              <string notr="true">#ReaderDetailsTitle
{
	font: bold 14px &quot;宋体&quot;;
}</string>
             </property>
             <property name="text">
              <string>读者信息</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="RefreshReaderDetails">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>15</width>
               <height>15</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>15</width>
               <height>15</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">#RefreshReaderDetails
{
	border: 0;
	image: url(:/Icon/refresh_white.png);
}
#RefreshReaderDetails:pressed
{
	image: url(:/Icon/refresh_black.png);
}</string>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="CloseDetails">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>15</width>
               <height>15</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>15</width>
               <height>15</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">#CloseDetails
{
	border:0;
	image: url(:/Icon/close_white.png);
}
#CloseDetails:pressed
{
	image: url(:/Icon/close_black.png);
}</string>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="ReaderDetailsContentLayout">
          <item>
           <widget class="QLabel" name="UserIcon">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>150</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">#UserIcon
{
	border-image: url(:/Icon/reader-black.png);
}</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="ReaderDetailsLayout">
            <property name="leftMargin">
             <number>15</number>
            </property>
            <item>
             <widget class="QWidget" name="UserIDWidget" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>30</height>
               </size>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLineEdit" name="UserID">
                 <property name="styleSheet">
                  <string notr="true">font-weight:bold;
font-size: 18px;</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <layout class="QGridLayout" name="ReaderDetailsGrid">
              <property name="horizontalSpacing">
               <number>25</number>
              </property>
              <property name="verticalSpacing">
               <number>15</number>
              </property>
              <item row="0" column="0">
               <widget class="QLabel" name="UserNameLabel">
                <property name="text">
                 <string>用户名</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="UserName"/>
              </item>
              <item row="0" column="3">
               <widget class="QLineEdit" name="College"/>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="PasswordLabel">
                <property name="text">
                 <string>密码</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="Password">
                <property name="echoMode">
                 <enum>QLineEdit::Password</enum>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="CollegeLabel">
                <property name="text">
                 <string>学院</string>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QComboBox" name="Permission"/>
              </item>
              <item row="1" column="2">
               <widget class="QLabel" name="PermissionLabel">
                <property name="text">
                 <string>借阅权限</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QWidget" name="widget" native="true">
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QPushButton" name="Save">
                 <property name="styleSheet">
                  <string notr="true">#Save
{
	width: 100px;
	height: 30px;
	border: 1px;
	border-radius: 12px;
	background-color: rgb(0, 120, 255);
	font-size: 16px;
	color: white;
}
#Save:hover
{
	background-color: rgb(0, 170, 255);
}
#Save:pressed
{
	border: 1px solid white;
	background-color:black;
	color: white;
}</string>
                 </property>
                 <property name="text">
                  <string>保存</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="Delete">
                 <property name="styleSheet">
                  <string notr="true">#Delete
{
	width: 100px;
	height: 30px;
	border: 1px;
	border-radius: 12px;
	background-color: rgb(198, 0, 0);
	font-size: 16px;
	color: white;
}
#Delete:hover
{
	background-color: rgb(220, 0, 0);
}
#Delete:pressed
{
	border: 1px solid white;
	background-color:black;
	color: white;
}</string>
                 </property>
                 <property name="text">
                  <string>删除读者</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="Status">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="ReadersListWidget" native="true">
       <property name="styleSheet">
        <string notr="true">#ReadersListWidget
{
	background-color: white;
	border: 1px solid rgb(235,235,235);
}</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QWidget" name="ReadersListTitleWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="ReadersListTitle">
             <property name="styleSheet">
              <string notr="true">#ReadersListTitle
{
	font: bold 14px &quot;宋体&quot;;
}</string>
             </property>
             <property name="text">
              <string>读者列表</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="RefreshReadersList">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>15</width>
               <height>15</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">#RefreshReadersList
{
	border: 0;
	image: url(:/Icon/refresh_white.png);
}
#RefreshReadersList:pressed
{
	image: url(:/Icon/refresh_black.png);
}</string>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="AddReader">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">#AddReader
{
	border: 0;
	color: rgb(154,168,172);
}
#AddReader:pressed
{
	color: black;
}</string>
             </property>
             <property name="text">
              <string>添加新读者</string>
             </property>
             <property name="icon">
              <iconset resource="resource.qrc">
               <normaloff>:/Icon/plus_white.png</normaloff>
               <normalon>:/Icon/plus_black.png</normalon>:/Icon/plus_white.png</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="SearchReaderLineWidget" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QWidget" name="SearchReader" native="true">
             <property name="styleSheet">
              <string notr="true">#SearchReader
{
	border: 0;
	border-bottom: 2px solid rgb(225,225,225);
}
QLineEdit, QPushButton
{
	font-size: 16px;
}</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <property name="spacing">
               <number>5</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="SearchType">
                <property name="styleSheet">
                 <string notr="true">#SearchType
{
	border: 0;
	color: gray;
}</string>
                </property>
                <property name="text">
                 <string>搜索类型</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="Keywords">
                <property name="styleSheet">
                 <string notr="true">#Keywords
{
	border: 0;
}</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="SearchButton">
                <property name="styleSheet">
                 <string notr="true">#SearchButton
{
	border:0;
	image:url(:/Icon/zoom.png);
	padding:5px;
	background-color:rgba(255,255,255,0);
}
#SearchButton:hover
{
	background-color:rgba(0,0,0,30);
}
#SearchButton:pressed
{
	background-color:rgba(0,0,0,60);
}</string>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="ReadersList">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="verticalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="horizontalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>AddReader</sender>
   <signal>clicked()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onAddClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>561</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Save</sender>
   <signal>clicked()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onSaveClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Delete</sender>
   <signal>clicked()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onDeleteClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CloseDetails</sender>
   <signal>clicked()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onCloseDetailsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>598</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SearchButton</sender>
   <signal>clicked()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onSearchButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>401</x>
     <y>335</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ReadersList</sender>
   <signal>cellClicked(int,int)</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onReaderSelected(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Keywords</sender>
   <signal>returnPressed()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>onSearchButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>330</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RefreshReadersList</sender>
   <signal>clicked()</signal>
   <receiver>ReadersWindow</receiver>
   <slot>listAllReaders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>498</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onAddClicked()</slot>
  <slot>onSaveClicked()</slot>
  <slot>onDeleteClicked()</slot>
  <slot>onSearchButtonClicked()</slot>
  <slot>onCloseDetailsClicked()</slot>
  <slot>onReaderSelected(int,int)</slot>
  <slot>listAllReaders()</slot>
 </slots>
</ui>
